File: dotnet/src/Shipzy/ValidationClient.cs
using System.Threading.Tasks;
using Shipzy.Sdk.Models;

namespace Shipzy.Sdk
{
    public class ValidationClient : ShipzyHttpClient
    {
        public ValidationClient(ZymeupConfig config) : base(config) { }

        public async Task<ApiResult<PhoneVerifyResult>> VerifyPhoneAsync(string countryCode, string phone)
            => await RequestAsync<ApiResult<PhoneVerifyResult>>("/api/v1/validation/phone", "POST", new { country_code = countryCode, phone });

        public async Task<ApiResult<PhoneFormatResult>> FormatPhoneAsync(string countryCode, string phone)
            => await RequestAsync<ApiResult<PhoneFormatResult>>("/api/v1/validation/phone/format", "POST", new { country_code = countryCode, phone });

        public async Task<ApiResult<PostalCodeResult>> ValidatePostalCodeAsync(string countryCode, string code)
            => await RequestAsync<ApiResult<PostalCodeResult>>("/api/v1/validation/postal-code", "POST", new { country_code = countryCode, code });

        public async Task<ApiResult<EmailValidationResult>> ValidateEmailAsync(string email)
            => await RequestAsync<ApiResult<EmailValidationResult>>("/api/v1/validation/email", "POST", new { email });

        public async Task<ApiResult<TaxIdValidationResult>> ValidateTaxIdAsync(string countryCode, string taxId)
            => await RequestAsync<ApiResult<TaxIdValidationResult>>("/api/v1/validation/tax-id", "POST", new { country_code = countryCode, tax_id = taxId });
    }
}

File: ruby/lib/shipzy/validation_client.rb
# frozen_string_literal: true

require_relative '../shipzy'

module Shipzy
  class ValidationClient < HttpClient
    def verify_phone(country_code, phone)
      request('/api/v1/validation/phone', method: :post, body: { country_code: country_code, phone: phone })
    end

    def format_phone(country_code, phone)
      request('/api/v1/validation/phone/format', method: :post, body: { country_code: country_code, phone: phone })
    end

    def validate_postal_code(country_code, code)
      request('/api/v1/validation/postal-code', method: :post, body: { country_code: country_code, code: code })
    end

    def validate_email(email)
      request('/api/v1/validation/email', method: :post, body: { email: email })
    end

    def validate_tax_id(country_code, tax_id)
      request('/api/v1/validation/tax-id', method: :post, body: { country_code: country_code, tax_id: tax_id })
    end
  end
end
